File: app/src/main/java/com/aquaguard/domain/model/ValveLog.java
package com.aquaguard.domain.model;

public class ValveLog {
    private final String logId;
    private final String deviceId;
    private final String triggeredBy;
    private final String action;
    private final String reason;
    private final long timestamp;

    public ValveLog(String logId, String deviceId, String triggeredBy, String action, String reason, long timestamp) {
        this.logId = logId != null ? logId : "";
        this.deviceId = deviceId != null ? deviceId : "";
        this.triggeredBy = triggeredBy != null ? triggeredBy : "";
        this.action = action != null ? action : "";
        this.reason = reason != null ? reason : "";
        this.timestamp = timestamp;
    }

    public String getLogId() { return logId; }
    public String getDeviceId() { return deviceId; }
    public String getTriggeredBy() { return triggeredBy; }
    public String getAction() { return action; }
    public String getReason() { return reason; }
    public long getTimestamp() { return timestamp; }
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/model/ValveLog.kt
package com.aquaguard.domain.model

data class ValveLog(
    val logId: String = "",
    val deviceId: String = "",
    val triggeredBy: String = "", // "USER" or "SYSTEM_AUTO"
    val action: String = "",      // "OPEN" or "CLOSE"
    val reason: String = "",
    val timestamp: Long = 0L
)
